pkg/sdk/testint/authentication_policies_gen_integration_test.go

```diff
@@ -0,0 +1,330 @@
+		require.NoError(t, err)
+		assert.Contains(t, desc, sdk.AuthenticationPolicyDescription{Property: "MFA_AUTHENTICATION_METHODS", Value: "[PASSWORD, SAML]"})
+	})
+
+	t.Run("Alter - unset mfa enrollment", func(t *testing.T) {
+		req := defaultCreateRequest()
+		err := client.AuthenticationPolicies.Create(ctx, req)
+		t.Cleanup(cleanupAuthenticationPolicyProvider(req.GetName()))
+
+		alterErr := client.AuthenticationPolicies.Alter(ctx, sdk.NewAlterAuthenticationPolicyRequest(req.GetName()).
+			WithUnset(*sdk.NewAuthenticationPolicyUnsetRequest().WithMfaEnrollment(true)))
+		require.NoError(t, alterErr)
+
+		desc, err := client.AuthenticationPolicies.Describe(ctx, req.GetName())
+		require.NoError(t, err)
+		assert.Contains(t, desc, sdk.AuthenticationPolicyDescription{Property: "MFA_ENROLLMENT", Value: "OPTIONAL"})
+	})
+
+	t.Run("Alter - unset comment", func(t *testing.T) {
+		req := defaultCreateRequest()
+		err := client.AuthenticationPolicies.Create(ctx, req)
+		t.Cleanup(cleanupAuthenticationPolicyProvider(req.GetName()))
+
+		alterErr := client.AuthenticationPolicies.Alter(ctx, sdk.NewAlterAuthenticationPolicyRequest(req.GetName()).
+			WithUnset(*sdk.NewAuthenticationPolicyUnsetRequest().WithComment(true)))
+		require.NoError(t, alterErr)
+
+		desc, err := client.AuthenticationPolicies.Describe(ctx, req.GetName())
+		require.NoError(t, err)
+		assert.Contains(t, desc, sdk.AuthenticationPolicyDescription{Property: "COMMENT", Value: "null"})
+	})
+
+	t.Run("Alter - rename", func(t *testing.T) {
+		req := defaultCreateRequest()
+		client.AuthenticationPolicies.Create(ctx, req)
+		t.Cleanup(cleanupAuthenticationPolicyProvider(req.GetName()))
+
+		newId := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		t.Cleanup(cleanupAuthenticationPolicyProvider(newId))
+		alterErr := client.AuthenticationPolicies.Alter(ctx, sdk.NewAlterAuthenticationPolicyRequest(req.GetName()).
+			WithRenameTo(newId))
+		require.NoError(t, alterErr)
+
+		_, descErr := client.AuthenticationPolicies.Describe(ctx, req.GetName())
+		assert.ErrorIs(t, descErr, sdk.ErrObjectNotExistOrAuthorized)
+	})
+
+	t.Run("Drop: existing", func(t *testing.T) {
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+
+		err := client.AuthenticationPolicies.Create(ctx, sdk.NewCreateAuthenticationPolicyRequest(id))
+		require.NoError(t, err)
+
+		err = client.AuthenticationPolicies.Drop(ctx, sdk.NewDropAuthenticationPolicyRequest(id))
+		require.NoError(t, err)
+
+		_, err = client.AuthenticationPolicies.ShowByID(ctx, id)
+		assert.ErrorIs(t, err, collections.ErrObjectNotFound)
+	})
+
+	t.Run("Drop: non-existing", func(t *testing.T) {
+		err := client.AuthenticationPolicies.Drop(ctx, sdk.NewDropAuthenticationPolicyRequest(NonExistingSchemaObjectIdentifier))
+		assert.ErrorIs(t, err, sdk.ErrObjectNotExistOrAuthorized)
+	})
+
+	t.Run("Show", func(t *testing.T) {
+		authenticationPolicy1 := createAuthenticationPolicy(t)
+		authenticationPolicy2 := createAuthenticationPolicy(t)
+
```

Comment:
the only thing I would add is a test with show filtering options to see which ones are available (even though they're not visible in the documentation yet; just use the ones you generated).

Reply:
Addressed in https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3068